extensions/pod-menu/src/attach-menu.tsx

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+
+
+import React from "react";
+import { Renderer, Common } from "@k8slens/extensions";
+
+type Pod = Renderer.K8sApi.Pod;
+
+const {
+  Component: {
+    createTerminalTab,
+    terminalStore,
+    MenuItem,
+    Icon,
+    SubMenu,
+    StatusBrick,
+  },
+  Navigation,
+} = Renderer;
+const {
+  Util,
+} = Common;
+
+export interface PodAttachMenuProps extends Renderer.Component.KubeObjectMenuProps<Pod> {
+}
+
+export class PodAttachMenu extends React.Component<PodAttachMenuProps> {
+  async attachToPod(container?: string) {
+    Navigation.hideDetails();
```

Comment:
Details should only be hidden if the rest of this function succeeds. I would recommend moving it to the bottom of this function.

Reply:
I based this behavior on the shell menu. We should propably change it there too, shouldn't we? https://github.com/lensapp/lens/blob/2bc56a038303ffd9c0f2bbc5cd0d8a667317543f/extensions/pod-menu/src/shell-menu.tsx#L47-L50

Reply:
Oh probably, but that should be a separate PR.

Reply:
Changed it.